pkg/kubelet/kubelet.go

```diff
@@ -2411,6 +2415,22 @@ func (kl *Kubelet) fastStatusUpdateOnce() {
 	}
 }
 
+func (kl *Kubelet) CheckpointContainer(podUID types.UID, podFullName, containerName string) error {
+	container, err := kl.findContainer(podFullName, podUID, containerName)
+	if err != nil {
+		return err
+	}
+	if container == nil {
+		return fmt.Errorf("container %v not found", containerName)
+	}
+
+	if err := kl.containerRuntime.CheckpointContainer(string(container.ID.ID), kl.getCheckpointsDir()); err != nil {
```

Comment:
> For example, here I expected that this path would include something like the pod namespace and/or pod id.

The path for the resulting checkpoint archive location? pod namespace and/or pod id could be added, but the container ID is unique, so there will be a way to figure out the origin of the checkpointed container. If this is what you meant it can be added if necessary. The plan for the checkpoint request is to pass additional information about the checkpoint creation via POST, for this PR I tried to keep it really simple as additional parameters can be added later.

Reply:
> The path for the resulting checkpoint archive location? pod namespace and/or pod id could be added, but the container ID is unique, so there will be a way to figure out the origin of the checkpointed container. 

It's unique but can be reused. The pod may be gone by the time the user goes to look for the checkpoint file.

> The plan for the checkpoint request is to pass additional information about the checkpoint creation via POST, for this PR I tried to keep it really simple as additional parameters can be added later.

right and we can also add more context later, and align that with the context needed for pod checkpointing

Reply:
Right, makes sense. I will change the name of the checkpoint archive.